pkg/api/ref/ref.go

```diff
@@ -63,11 +63,11 @@ func GetReference(scheme *runtime.Scheme, obj runtime.Object) (*api.ObjectRefere
 
 	// An object that implements only List has enough metadata to build a reference
 	var listMeta metav1.Common
-	objectMeta, err := meta.Accessor(obj)
-	if err != nil {
-		listMeta, err = meta.CommonAccessor(obj)
-		if err != nil {
-			return nil, err
+	objectMeta, ok := meta.Accessor(obj)
+	if !ok {
+		listMeta, ok = meta.CommonAccessor(obj)
+		if !ok {
+			return nil, fmt.Errorf("object does not implement the common interface for accessing the SelfLink")
```

Comment:
why is it better to scatter construction of this error?

Reply:
I notice there are TODOs in apimachinery\pkg\api\meta\meta.go,return bool instead of error func CommonAccessor, ListAccessor, Accessor. and i open a issue for it #61153

Reply:
after seeing the alternative (constructing essentially the same error in lots of places), I think I prefer removing the TODO

Reply:
@liggitt I have re pushed it, remove the TODOs, Thanks